source_code/main_mcu/src/GUI/gui_menu.c

```diff
@@ -293,12 +293,54 @@ BOOL gui_menu_event_render(wheel_action_ret_te wheel_action)
                         else if (display_prompt_return == MINI_INPUT_RET_YES)
                         {
                             nodemgmt_read_cred_child_node(chosen_login_addr, (child_cred_node_t*)&temp_cnode);
-                            logic_gui_display_login_password_TOTP((child_cred_node_t*)&temp_cnode);
+                            logic_gui_display_login_password_TOTP((child_cred_node_t*)&temp_cnode, FALSE);
                             memset(&temp_cnode, 0, sizeof(temp_cnode));
                             return TRUE;
                         }
                         else
                         {
+                            // Read node so that we can check if TOTP is available
+                            nodemgmt_read_cred_child_node(chosen_login_addr, (child_cred_node_t*)&temp_cnode);
+
+                            if (((child_cred_node_t *) &temp_cnode)->TOTP.TOTPsecretLen > 0)
+                            {
+                                // Fetch parent node to prepare prompt text
+                                temp_pnode_pt = (parent_node_t*)&temp_cnode;
+                                nodemgmt_read_parent_node(chosen_service_addr, temp_pnode_pt, TRUE);
+
+                                // Ask the user if he wants to display TOTP on screen
+                                cust_char_t* display_totp_prompt_text;
+                                custom_fs_get_string_from_file(QPROMPT_SNGL_DISP_TOTP_TEXT_ID, &display_totp_prompt_text, TRUE);
+                                confirmationText_t totp_prompt_object = {.lines[0] = temp_pnode_pt->cred_parent.service, .lines[1] = display_totp_prompt_text};
+                                display_prompt_return = gui_prompts_ask_for_confirmation(2, &totp_prompt_object, FALSE, TRUE, FALSE);
+
+                                if (display_prompt_return == MINI_INPUT_RET_BACK)
+                                {
+                                    if ((logic_bluetooth_get_state() != BT_STATE_CONNECTED) && (logic_aux_mcu_is_usb_enumerated() == FALSE))
+                                    {
+                                        state_machine_index = 0;
+                                    }
+                                    else
+                                    {
+                                        only_password_prompt = TRUE;
+                                        state_machine_index--;
```

Comment:
Imagine the following scenario: the user wants to display the favorite, but for some reason makes a mistake and denies the prompt. He's then prompted for TOTP only display, long clicks to go back but instead of getting prompted for credential display, he's getting prompted to type credentials